java/src/main/java/com/powsybl/python/network/NetworkModificationCFunctions.java

```diff
@@ -0,0 +1,100 @@
+
+import java.util.*;
+
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.*;
+import static com.powsybl.python.commons.Util.createIntegerArray;
+import static com.powsybl.python.commons.Util.doCatch;
+
+/**
+ * @author Coline Piloquet <coline.piloquet at rte-france.com>
+ */
+@CContext(Directives.class)
+public final class NetworkModificationCFunctions {
+
+    private NetworkModificationCFunctions() {
+    }
+
+    @CEntryPoint(name = "getConnectablesOrderPositions")
+    public static PyPowsyblApiHeader.ArrayPointer<PyPowsyblApiHeader.SeriesPointer> getConnectablesOrderPositions(IsolateThread thread, ObjectHandle networkHandle,
+                                                     CCharPointer voltageLevelId, PyPowsyblApiHeader.ExceptionHandlerPointer exceptionHandlerPtr) {
+        return doCatch(exceptionHandlerPtr, () -> {
+            String voltageLevelIdStr = CTypeUtil.toString(voltageLevelId);
+            Network network = ObjectHandles.getGlobal().get(networkHandle);
+
+            VoltageLevel voltageLevel = network.getVoltageLevel(voltageLevelIdStr);
+            Map<String, List<Integer>> feederPositionsOrders = getFeederPositionsByConnectable(voltageLevel);
+            //TODO: Make it better. For now there's nothing to create a dataframe with a list as a column
+            Map<String, String> feederPositionsOrdersString = new HashMap<>();
+            feederPositionsOrders.forEach((k, v) -> feederPositionsOrdersString.put(k, v.toString()));
+            return Dataframes.createCDataframe(Dataframes.mapMapper(), feederPositionsOrdersString);
+        });
+
+    }
+
+    @CEntryPoint(name = "getUnusedConnectableOrderPositionsBefore")
+    public static PyPowsyblApiHeader.ArrayPointer<CIntPointer> getUnusedConnectableOrderPositionsBefore(IsolateThread thread, ObjectHandle networkHandle,
+                                                                                             CCharPointer busbarSectionId,
+                                                                                             PyPowsyblApiHeader.ExceptionHandlerPointer exceptionHandlerPtr) {
+        Network network = ObjectHandles.getGlobal().get(networkHandle);
+        String busbarSectionIdStr = CTypeUtil.toString(busbarSectionId);
+        BusbarSection busbarSection = network.getBusbarSection(busbarSectionIdStr);
+        try {
+            Optional<Range<Integer>> positionsOrders = getUnusedOrderPositionsBefore(busbarSection);
+            if (positionsOrders.isPresent()) {
+                int max = positionsOrders.get().getMaximum();
+                int min = positionsOrders.get().getMinimum();
+                return createIntegerArray(Arrays.asList(min, max));
+            } else {
+                return createIntegerArray(Collections.EMPTY_LIST);
+            }
+        } catch (PowsyblException e) {
+            throw new PowsyblException(e.getMessage());
+        }
```

Comment:
Here and on the method afterwards, it seems like the exceptions are not caught...

Reply:
is it still the case ?

Reply:
just maybe both methods can be merged into one they seemed to be very similar